srv/salt/ceph/osd/takeover/default.sls

```diff
@@ -0,0 +1,16 @@
+{% if 'storage' in salt['pillar.get']('roles') %}
+
+{% for id in salt['osd.list']() %}
+taking over osd {{ id }}:
+  cmd.run:
+    - name: "ceph-volume simple scan /var/lib/ceph/osd/ceph-{{ id }}"
```

Comment:
It appears that it's not enough to run this command..

c-v gives you a command to 'activate' after a successful 'simple scan'.

```
x@x: ceph-volume simple scan /dev/vdb1

Running command: /usr/sbin/cryptsetup status /dev/vdb1
--> OSD 1 got scanned and metadata persisted to file: /etc/ceph/osd/1-ccc15e6f-37bd-48dc-af22-70ba053bc6c3.json
--> To take over management of this scanned OSD, and disable ceph-disk and udev, run:
-->     ceph-volume simple activate 1 ccc15e6f-37bd-48dc-af22-70ba053bc6c3
```

Since this is currently running in a sls file we don't have a chance to pick this line up or compute the fsid..

We may have to switch to a python module for this.

Reply:
A single command for this would be useful. Do we have enough time though?

Reply:
you can scan all OSDs and then run `ceph-volume simple activate --all`

Reply:
did you happen to check if the scan is idempotent? I seem to remember that it is, but best double check if we might need a conditional.

Reply:
> did you happen to check if the scan is idempotent? I seem to remember that it is, but best double check if we might need a conditional.

it's not. but adding a `--force` works as the content doesn't change

Reply:
> you can scan all OSDs and then run ceph-volume simple activate --all

neat! I must have overlooked that..

Reply:
> it's not. but adding a `--force` works as the content doesn't change

hmm or we check for the created json file with a `unless` prerequisite...

Reply:
The thing is that we don't know the second param (osd_fsid) to construct the json file.

> A previously scanned OSD would exist like::

    /etc/ceph/osd/{ID}-{FSID}.json

if we could extract it easily, we also don't have to do the `activate --all`

the scan/activate wasn't designed with automation in mind as it seems

Reply:
Oh, and at least in my version (

> admin:~ # ceph-volume simple activate --all
usage: ceph-volume simple activate [-h] [--file FILE] [--no-systemd]
                                   [ID] [FSID]
ceph-volume simple activate: error: unrecognized arguments: --all